lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -49,6 +49,52 @@ DenseElementsAttr getConstantValue(Builder &builder, Attribute value,
   return res;
 }
 
+bool isAddPtrOffsetCombinable(Value first, Value second) {
+  auto GetConstantIntValue = [](Value val) -> std::optional<llvm::APInt> {
+    DenseElementsAttr constAttr;
+    Attribute attr;
+    auto defOp = val.getDefiningOp();
+    if (auto splatOp = llvm::dyn_cast_or_null<SplatOp>(defOp))
+      val = splatOp.getSrc();
+    else if (matchPattern(defOp, m_Constant(&constAttr)) && constAttr.isSplat())
+      attr = constAttr.getSplatValue<Attribute>();
+
+    // Check constant value.
+    llvm::APInt intVal;
+    if (matchPattern(val, m_ConstantInt(&intVal)))
+      return intVal;
+
+    // Check IntegerAttr
+    if (auto intAttr = dyn_cast_or_null<IntegerAttr>(attr))
+      return intAttr.getValue();
+
+    return std::nullopt;
+  };
+
+  Type elem1Type = first.getType();
+  Type elem2Type = second.getType();
+  if (elem1Type.isa<ShapedType>() && elem2Type.isa<ShapedType>()) {
+    elem1Type = elem1Type.cast<ShapedType>().getElementType();
+    elem2Type = elem2Type.cast<ShapedType>().getElementType();
+  }
```

Comment:
I mentioned it before, I don't understand why we need to get the element type, checking `elem1Type == elem2Type` should be enough for tensors as well.